zaza/model.py

```diff
@@ -1013,6 +1024,16 @@ async def async_wait_for_application_states(model_name=None, states=None,
                             unit_state=unit.workload_status_message,
                             approved_states=prefixes))
 
+                logging.info("Waiting for application to be active")
+                try:
+                    await model.block_until(
+                        lambda: check_applications_status(app_data,
+                                                          approved_statuses),
```

Comment:
There doesn't appear to by anything here that will change; i.e. the lambda is capturing static structures and thus there's no change to detect?  (unless I'm missing something obvious, which may well be the case).

i.e. check_applications_status(x, y) doesn't get new information the way check_applications_status(..) is written - there's no "get new status info" in the check.